crates/oxc_transformer/src/jsx/comments.rs

```diff
@@ -96,20 +98,26 @@ enum PragmaType {
 fn find_jsx_pragma(mut comment_str: &str) -> Option<(PragmaType, &str, &str)> {
     let pragma_type;
     loop {
-        // Search for `@jsx`.
-        let mut at_sign_index = None;
-        for (index, next4) in comment_str.as_bytes().windows(4).enumerate() {
-            if next4 == b"@jsx" {
-                at_sign_index = Some(index);
-                break;
-            }
+        // Search for `@`.
+        // Note: Using `memchr::memmem::Finder` to search for `@jsx` is slower than only using `memchr`
+        // to find `@` characters, and then checking if `@` is followed by `jsx` separately.
+        let at_sign_index = memchr(b'@', comment_str.as_bytes())?;
```

Comment:
it might also be faster to use memchr_iter rather than a loop and having the searcher re-compiled again. (it also might make no difference depending on how the compiler is optimizing it)

Reply:
I don't think `memchr` (unlike `memchr::memmem::Finder`) compiles a searcher type - it's a straight function call. If it's inlined, the "needle" (`@`) should be statically known to compiler, which might allow it to generate more optimal code than a more complex type.

I've taken some shortcuts here - there are some bounds checks which could be eliminated, and other stuff that could be a little tighter. But my assumption was that almost all the cost is in searching for `@`, so everything after that was not worthwhile optimizing.

But still, yes, `memchr_iter` might be faster, especially if the searcher was created outside of `find_jsx_pragma` and passed in so it can be re-used. Want to try it?